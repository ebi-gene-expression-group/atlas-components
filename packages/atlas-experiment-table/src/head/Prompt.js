import React from 'react'
import PropTypes from 'prop-types'

class Prompt extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      selectedFileTypes: props.downloadFileTypes.map(fileType => fileType.id)
    }

    this.onClick = this.onClick.bind(this)
  }

  onClick(selectedFileType) {
    const selectedFileTypes = [...this.state.selectedFileTypes]
    this.setState({
      selectedFileTypes: selectedFileTypes.includes(selectedFileType) ?
        selectedFileTypes.filter(fileType => fileType !== selectedFileType) :
        [selectedFileType, ...selectedFileTypes]
      },
      () => this.props.onSelect(this.state.selectedFileTypes)
    )
  }

  render() {
    return <div>
        <p style={{paddingBottom: `1rem`}}>Choose file types...</p>
          {
            this.props.downloadFileTypes.map((fileType, idx) =>
              <div key={`filetype${idx}`} style={{display:"flex"}}>
                <input type={`checkbox`} className={`checkbox`} id={fileType.id}
                       checked={this.state.selectedFileTypes.includes(fileType.id)}
                       onChange={() => this.onClick(fileType.id)}/>
                <p id={fileType.description} style={{paddingLeft: `1rem`, paddingBottom: `1rem`}}>
                  {fileType.description}
                </p>
              </div>
            )
          }

      </div>

  }
}

Prompt.propTypes = {
  onSelect: PropTypes.func.isRequired,
  downloadFileTypes: PropTypes.arrayOf(
    PropTypes.shape({
      description: PropTypes.string,
      id: PropTypes.string
    })
  ).isRequired
}

export default Prompt
