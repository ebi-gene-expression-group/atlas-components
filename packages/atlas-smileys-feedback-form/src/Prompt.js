import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import smileyData from './smileyData'
import SmileyDiv from './SmileyDiv'

const SmileyContainer = styled.div`
  display: flex;
  justify-content: space-evenly;
`

class Prompt extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      selectedSmileyScore: 0
    }

    this.onClick = this.onClick.bind(this)
  }

  onClick(selectedSmileyScore) {
    this.setState({
      selectedSmileyScore: selectedSmileyScore
    })
    this.props.onSelect(selectedSmileyScore)
  }

  render() {
    return(
      <div>
        <p style={{paddingBottom: `1rem`}}>How satisfied are you?</p>
        <SmileyContainer>
          {
            smileyData.map((smiley, idx) =>
              <SmileyDiv key={idx} onClick={() => this.onClick(smiley.score)}
                emoji={smiley.emoji}
                label={smiley.label}
                selected={smiley.score === this.state.selectedSmileyScore}/>
            )
          }
        </SmileyContainer>
        <p style={{paddingTop: `1rem`}}>
          <a href={this.props.feedbackFormLink} target={`_blank`}>Click here if you need support.</a>
        </p>
      </div>
    )
  }
}

Prompt.propTypes = {
  onSelect: PropTypes.func.isRequired,
  feedbackFormLink: PropTypes.string.isRequired
}

export default Prompt
