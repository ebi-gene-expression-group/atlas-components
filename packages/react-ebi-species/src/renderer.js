"use strict";

//*------------------------------------------------------------------*

var React = require('react');
var ReactDOM = require('react-dom');

//*------------------------------------------------------------------*

var Icon = require('./SpeciesIcon.jsx');

//*------------------------------------------------------------------*

module.exports = function(mountNode,species,colourOverride, colourPerKingdom) {
    ReactDOM.render(
        React.createElement(Icon,{
          species: species,
          colourOverride: colourOverride,
          colourPerKingdom: colourPerKingdom
        }), mountNode
    );
};
