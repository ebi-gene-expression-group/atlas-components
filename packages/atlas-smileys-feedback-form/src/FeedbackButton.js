import React from 'react'
import PropTypes from 'prop-types'

import styled from 'styled-components'
import Popup from 'react-popup'
import ReactGA from 'react-ga'

import Prompt from './Prompt'

const FeedbackButtonDiv = styled.button`
  position: fixed;
  bottom: 50%;
  right: 0px;
  display: inline-block;
  border-radius: 0px;
  background-color: #3497c5;
  border: none;
  color: white;
  text-align: center;
  font-size: 1.2rem;
  padding: 0.25rem;
  transition: all 0.2s;
  margin: -40px;
  &:hover {
    background-color: #2f5767;
    cursor: pointer;
  }
  box-shadow: 2px 2px 3px #999;
  -ms-transform: rotate(-90deg); /* IE 9 */
  -webkit-transform: rotate(-90deg); /* Safari 3-8 */
  transform: rotate(-90deg);
`

class FeedbackButton extends React.Component{

  constructor(props) {
    super(props)

    this.state = {
      smiley: 0
    }
    this.onClick = this.onClick.bind(this)
    this.smileyChange = this.smileyChange.bind(this)
  }

  smileyChange(smileyValue){
    this.setState(
      {smiley: smileyValue},
      () => {
        Popup.close() 
        this.onClick()
      })
  }

  onClick(){
    const {feedbackFormLink} = this.props

    Popup.registerPlugin(`prompt`, function (smiley, smileyChange, callback) {
      this.create({
        title: `Your feedback`,
        content: <Prompt feedbackFormLink={feedbackFormLink} onSelect={smileyChange} />,
        buttons: {
          left: [`cancel`],
          right: smiley ? [{
            text: `Save`,
            className: `success`,
            action: function () {
              callback()
              smiley && ReactGA.event({
                category: `Satisfaction`,
                action: smiley.toString()
              })
              smiley && Popup.close()
            }
          }] : []
        }
      })
    })

    Popup.plugins().prompt(this.state.smiley, this.smileyChange, function () {
      Popup.alert(`Thank you for submitting your feedback.`)
    })

  }

    ReactGA.initialize(this.props.gaId)
    ReactGA.pageview(window.location.pathname + window.location.search)

    return(
      <div>
        <Popup defaultOk={`OK`} />
        <FeedbackButtonDiv id={`feedback-button`} onClick={this.onClick}>
          <i className={`icon icon-functional`} data-icon="n"> </i>Feedback
        </FeedbackButtonDiv>
      </div>
    )
  }
}

FeedbackButton.propTypes = {
  feedbackFormLink: PropTypes.string.isRequired,
  gaId: PropTypes.string.isRequired
}

export default FeedbackButton
