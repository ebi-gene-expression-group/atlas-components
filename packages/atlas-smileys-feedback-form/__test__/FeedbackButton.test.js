import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import {shallow, mount} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Prompt from '../src/Prompt'
import FeedbackButton from '../src/FeedbackButton.js'

jest.mock("react-ga")

Enzyme.configure({ adapter: new Adapter() })

describe(`FeedbackButton`, () => {
  test(`should render a side feedback button`, () => {
    expect(shallow(<FeedbackButton feedbackFormLink={`linto`} GAid={`id`}/>).find(`#feedback-button`)).toHaveLength(1)
  })

  test(`should popup a feedback form by clicking a button`, () => {
    const wrapper = mount(<FeedbackButton feedbackFormLink={`linto`} GAid={`id`}/>)
    expect(wrapper.find(Prompt)).toHaveLength(0)
    wrapper.find(`button`).simulate('click')
    expect(wrapper.find(Prompt)).toHaveLength(1)
  })

  test(`matches snapshot`, () => {
    const tree = renderer.create(<FeedbackButton feedbackFormLink={`linto`} GAid={`id`}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
