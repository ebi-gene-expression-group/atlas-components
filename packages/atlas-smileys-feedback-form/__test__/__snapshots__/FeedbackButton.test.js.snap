// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackButton matches snapshot 1`] = `
<div>
  <div
    className="mm-popup"
  >
    <div
      className="mm-popup__overlay"
      onClick={[Function]}
      role="presentation"
      style={Object {}}
    />
  </div>
  <button
    className="sc-htpNat cvazSS"
    onClick={[Function]}
  >
    <i
      className="icon icon-functional"
      data-icon="n"
      style={
        Object {
          "paddingRight": "0.5rem",
        }
      }
    />
    Feedback
  </button>
</div>
`;
